### DISCUSSION TITLE
PA1010D GPS driver in micropython

### DISCUSSION THREAD

**Discussion by arobertsonithree** [micropython/micropython#11694] [Libraries & Drivers]
I've tried to get a PA1010D GPS working but failed to find a driver to suit. For various reasons it needs to be micropyhon (not circuitpython) and use a I2C interface. I've found the circuitpython one and Pimoroni publish a python one (https://github.com/pimoroni/pa1010d-python). But I have can't get it to work. ChatGPT is absolutely useless in converting these to micropython. Does anyone know of a good native micropython driver for I2C I could try? I'm not an expert in this so may have missed something obvious.

**Comment by 2dof** [comment]
Go to  https://github.com/orgs/micropython/discussions/11651 , user   @divergentti  implemented driver for [NEO-6M GPS](https://github.com/divergentti/Micropython-for-ESP32/blob/main/esp32-gps/GPS_AS.py)  at least for  MTK NMEA Packet Protocol  you will be able to adapt code, or just adapt [adafruit_gps](https://github.com/adafruit/Adafruit_CircuitPython_GPS/blob/main/adafruit_gps.py)

For connection/communication with  PA1010D GPS, standard  (native) micro-python i2C will be sufficient, you need just read message or send command.  

 edit: on [awesome-micropython](https://awesome-micropython.com/ )  you will find some links to gps libs , for example [micropyGPS](https://github.com/inmcm/micropyGPS ) it is for pyboard.
